test

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+import os
+from subprocess import run, PIPE, Popen
+from pathlib import Path, PurePath
+from distutils.spawn import find_executable
+from optparse import OptionParser
+
+class Checker(object):
+  checked = []
+
+  def __init__(self, compiler, file):
+    self.compiler = compiler.split()
+    self.file = file
+    self.check()
+
+  def check(self):
+    binary = self.compiler[0]
+
+    if binary not in self.checked and not find_executable(binary):
+      raise EnvironmentError("{!r} not found. Do you have the compilers?".format(binary))  # noqa
+    elif binary not in self.checked:
+      self.checked += binary
+
+class Execute(Checker):
+  def run(self, input):
+    command = ' '.join(self.compiler + [self.file])
+    p = Popen(command, stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
+
+    return p.communicate(input=input.encode())
+
+class Build(Checker):
+  fout = "compiled.out"
+
+  def compile(self):
+    args = self.compiler + [self.file, "-o", self.output]
+    program = Popen(args, stdout=PIPE)
+
+    return program.wait() == 0
+
+  def run(self, input):
+    self.output = os.path.join(os.path.dirname(self.file), self.fout)
+
+    if self.compile():
+      compiled = os.path.abspath(self.output)
+      program = Execute("bash -c", "{!r}".format(compiled))
+      output = program.run(input)
+      os.remove(compiled)
+      return output
+
+    return b"compiles fails", EnvironmentError, 0
+
+BUILD_MACHINE = {
```

Comment:
Well... This is really bad. I think we can keep this as is it. In another hour we can improve that.